builds/e2e/nested-connectivity.yaml

```diff
@@ -31,12 +31,20 @@ resources:
     branch: 'release/1.2'
 
 stages:
-- stage: SetupVMs
+- stage: LockAgents 
+  jobs:
+  - template: templates/lock-test-agents.yaml
+    parameters:
+      upstream.protocol: 'mqtt'
+      testRunnerCount: 2
```

Comment:
For this lock test agent functionality, I took logic out of parent-vm-setup and made a dedicated yaml. This better facilitates splitting the stages into:
1. Setup
2. Run Tests
3. Cleanup